### PR TITLE
APS-760:Add validation to index offence selection screen

### PR DESCRIPTION
**Pull request by vubalasu**
# Context

# Changes in this PR
Add validation to index offence selection screen
### Before
image

### After
Screenshot 2024-05-30 at 14 50 12

### CODE REVIEW

File: server/controllers/apply/applicationsController.ts [hunk 1]
```diff
@@ -165,7 +166,14 @@ export default class ApplicationsController {
     return async (req: Request, res: Response) => {
       const { crn, offenceId } = req.body
 
+      const person = await this.personService.findByCrn(req.user.token, crn)
+      if (!isFullPerson(person)) throw new RestrictedPersonError(crn)
       const offences = await this.personService.getOffences(req.user.token, crn)
+      if (!offenceId) {
+        addErrorMessageToFlash(req, 'You must select the offence', 'offenceId')
```
Comment (edavey) [inline server/controllers/apply/applicationsController.ts:173]: perhaps 

> You must select the index offence

would be more accurate?

File: server/controllers/apply/people/offencesController.test.ts [hunk 2]
```diff
@@ -45,6 +47,8 @@ describe('OffencesController', () => {
 
       expect(response.render).toHaveBeenCalledWith('applications/people/selectOffence', {
         pageHeading: `Select index offence for ${person.name}`,
+        errors: {},
+        errorSummary: [],
```
Comment (edavey) [inline server/controllers/apply/people/offencesController.test.ts:51]: are we actually describing the unhappy path as well as the happy path?

I was expecting to see a scenario along the lines of 

```ts
expect(response.render).toHaveBeenCalledWith('applications/people/selectOffence', {
        ...
        errors: errorsAndUserInput.errors,
        errorSummary: errorsAndUserInput.errorSummary,
        ...errorsAndUserInput.userInput,
      })
```

**Review by edavey — COMMENTED** [review]
Thanks for this. A couple of comments:

- a minor suggestion on the error message and 
- not sure that the error case is clearly described in the controller test?

File: server/controllers/apply/applicationsController.ts (hunk 1 above)
Comment (vubalasu) [reply server/controllers/apply/applicationsController.ts]: updated

File: server/controllers/apply/people/offencesController.test.ts (hunk 2 above)
Comment (vubalasu) [reply server/controllers/apply/people/offencesController.test.ts:52]: @edavey added

**Review by edavey — APPROVED** [review]
thanks for the tweaks

**Pull request merged**